src/Framework/IBuildEngineInternal.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Build.Framework
+{
+    /// <summary>
+    ///
+    /// </summary>
+    internal interface IBuildEngineInternal
+    {
+        /// <summary>
+        /// Returns the minimum message importance not guaranteed to be ignored by registered loggers.
+        /// </summary>
+        /// <remarks>
+        /// Example: If we know that no logger is interested in MessageImportance.Low, this property returns
+        /// MessageImportance.Normal. If loggers may consume any messages, this property returns MessageImportance.Low.
+        /// </remarks>
+        MessageImportance MinimumRequiredMessageImportance { get; }
```

Comment:
Isn't it more fluent to have users call it like `if (BuildEngine.LogsMessagesOfImporance(MessageImportance.Low)) {/*log low importance message*/}`?

Reply:
I've turned it into a method as suggested: `bool LogsMessagesOfImportance(MessageImportance importance)`, thank you!